### PR TITLE
[MU4] Concept: keyboard navigation by subsections

### PR DESCRIPTION
**Pull request by igorkorsukov**
including https://github.com/musescore/MuseScore/pull/7904

https://user-images.githubusercontent.com/3818029/113729132-0468d600-96f7-11eb-944e-8c77af54323c.mp4

### CODE REVIEW

File: src/framework/ui/ikeynavigationsection.h [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#ifndef MU_UI_IKEYNAVIGATIONSECTION_H
+#define MU_UI_IKEYNAVIGATIONSECTION_H
+
+#include <QString>
+#include <QList>
+
+namespace mu::ui {
+class IKeyNavigationSubSection
```
Comment (RomanPudashkin) [inline src/framework/ui/ikeynavigationsection.h:26]: Do we really need this interface? It seems that we only need IKeyNavigationSection. Technically a situation may happen when we need sub-sub sections and I afraid that we'll create IKeyNavigationSubSubSection :)

Comment (igorkorsukov) [reply src/framework/ui/ikeynavigationsection.h:26]: I thought :) And I decided to add a separate interface, for two purposes:
1. Avoid `section-> section-> section...` tree. We only have a `section-> subsection-> control` and that's it.
2. For cleaning type

File: src/framework/ui/uitypes.h [hunk 2]
```diff
@@ -162,14 +162,16 @@ struct UiAction
     std::string shortcut;
 
     UiAction() = default;
-    UiAction(const actions::ActionCode& code, UiContext ctx, const QString& title, Checkable ch = Checkable::No)
+    UiAction(const actions::ActionCode& code, UiContext ctx, Checkable ch = Checkable::No)
+        : code(code), context(ctx), checkable(ch) {}
+    UiAction(const actions::ActionCode& code, UiContext ctx, const char* title, Checkable ch = Checkable::No)
         : code(code), context(ctx), title(title), checkable(ch) {}
-    UiAction(const actions::ActionCode& code, UiContext ctx, const QString& title, const QString& desc, Checkable ch = Checkable::No)
+    UiAction(const actions::ActionCode& code, UiContext ctx, const char* title, const char* desc, Checkable ch = Checkable::No)
```
Comment (RomanPudashkin) [inline src/framework/ui/uitypes.h:169]: Better to use std::string_view instead of const char*

Comment (igorkorsukov) [reply src/framework/ui/uitypes.h:169]: no, not in this case :)
typically we do
```
const UiActionList ApplicationUiActions::m_actions = {
    UiAction("quit",
             mu::context::UiCtxAny,
             QT_TRANSLATE_NOOP("action", "Quit")
             ),
    UiAction("fullscreen",
             mu::context::UiCtxAny,
             QT_TRANSLATE_NOOP("action", "Full Screen"),
             QT_TRANSLATE_NOOP("action", "Full screen"),
             Checkable::Yes
             ),
```
here `QT_TRANSLATE_NOOP("action", "Esc")` just `const char*`

File: src/framework/ui/ikeynavigationsection.h (hunk 1 above)
Comment (RomanPudashkin) [reply src/framework/ui/ikeynavigationsection.h:26]: Make sense, but I don't like that we have two almost identical interfaces. Maybe is it better to create one base interface and inherit IKeyNavigationSection from it?

Comment (igorkorsukov) [reply src/framework/ui/ikeynavigationsection.h:26]: I think this is overengineering :) [KISS](https://en.wikipedia.org/wiki/KISS_principle)

Comment (igorkorsukov) [reply src/framework/ui/ikeynavigationsection.h:26]: In the next PR I going to add common interface

**Review by RomanPudashkin — APPROVED** [review]

**Pull request merged**